woocommerce/woocommerce-google-analytics-integration
# Upgrade to Node.js 18 and bump all NPM dev dependencies

### Changes proposed in this Pull Request:

Closes #281 

- Upgrade to use Node.js v18.
- Bump all dev dependencies to solve all vulnerabilities.
   - This reduces the number of npm vulnerabilities from 34 to 0. 🎉 
- Fix JS linting errors.

### Checks:


### Detailed test instructions:

1. Set Node.js to v18.
2. Run `npm install` to see if it can finish without errors.
3. Run `npm audit` to check if it reports 0 vulnerabilities.
4. Check if `npm start`, `npm run dev`, `npm run build`, and `npm run lint:js` work well.
5. Go to the admin page of **WooCommerce > Settings > Integration > Google Analytics. Toggle the "Use Global Site Tag" checkbox to see if it can show/hide the "Set Domain Name" option.
6. Follow the same test steps in #343 to determine if the event tracking is sent with **Checkout** block.

### Changelog entry

- Dev - Upgrade to Node.js 18.
- Dev - Upgrade all NPM dev dependencies.
